perf-tool/src/methodEntry.cpp

```diff
@@ -33,65 +33,76 @@
+        char *name_ptr;
+        char *signature_ptr;
+        jvmtiError err = jvmtiEnv->GetMethodName(method, &name_ptr, &signature_ptr, NULL);
+        if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve Method Name.\n")) 
+        {
             j["methodName"] = name_ptr;
             j["methodSig"] = signature_ptr;
             err = jvmtiEnv->Deallocate((unsigned char*)name_ptr);
             check_jvmti_error(jvmtiEnv, err, "Unable to deallocate name_ptr.\n");
             err = jvmtiEnv->Deallocate((unsigned char*)signature_ptr);
             check_jvmti_error(jvmtiEnv, err, "Unable to deallocate signature_ptr.\n");
         }
+        else
+        {
+            /* Without the method name, there is no point in continuing to get the class */
+            return; 
+        }
+
+        jclass declaring_class;
         err = jvmtiEnv->GetMethodDeclaringClass(method, &declaring_class);
-        if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve Method Declaring Class.\n")) {
+        if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve Method Declaring Class.\n")) 
+        {
+            char *declaringClassName;
             err = jvmtiEnv->GetClassSignature(declaring_class, &declaringClassName, NULL);
-            if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve Method Declaring Class Signature.\n")) {
+            if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve Method Declaring Class Signature.\n")) 
+            {
                 j["methodClass"] = declaringClassName;
                 err = jvmtiEnv->Deallocate((unsigned char*)declaringClassName);
                 check_jvmti_error(jvmtiEnv, err, "Unable to deallocate declaringClassName.\n");
             }
         }
+
+        /* Get StackTrace */
+        methodEnterConfig.getStackTrace(jvmtiEnv, thread, j, methodEnterConfig.getStackTraceDepth());
+/*
+        The following code needs "can_get_line_numbers" capability.
+        Due to the high frequency of MethodEnter events it's better to keep
+        the overhead down by not getting information about the line numbers.
+        This can be added in the future through an addional option.
+
+        jint entry_count_ptr;
+        jvmtiLineNumberEntry* table_ptr;
         err = jvmtiEnv->GetLineNumberTable(method, &entry_count_ptr, &table_ptr);
-        if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve Method Line Number Table.\n")) {
+        if (check_jvmti_error(jvmtiEnv, err, "Unable to retrieve Method Line Number Table.\n")) 
+        {
             j["methodLineNum"] = table_ptr->line_number;
             err = jvmtiEnv->Deallocate((unsigned char*)table_ptr);
             check_jvmti_error(jvmtiEnv, err, "Unable to deallocate table_ptr.\n");
         }
-    
-        std::string s = j.dump();
-        sendToServer(s);
+*/
+        sendToServer(j.dump());
     }
-    mEntrySampleCount = atomic_fetch_add(&mEntrySampleCount, 1);

```

Comment:
Don't we need this? Otherwise `mEntrySampleCount` will never get incremented.

Reply:
There is another increment on line 45. This one was a duplicate.